executor/simple.go

```diff
@@ -588,6 +596,47 @@ func (e *SimpleExec) executeBegin(ctx context.Context, s *ast.BeginStmt) error {
 	return nil
 }
 
+func (e *SimpleExec) executeStartTransactionReadOnlyWithTimestampBound(ctx context.Context, s *ast.BeginStmt) error {
+	opt := sessionctx.StalenessTxnOption{}
+	opt.Mode = s.Bound.Mode
+	switch s.Bound.Mode {
+	case ast.TimestampBoundReadTimestamp:
+		v, ok := s.Bound.Timestamp.(*driver.ValueExpr)
+		if !ok {
+			return errors.New("Invalid timestamp")
+		}
+		t, err := types.ParseDatetime(e.ctx.GetSessionVars().StmtCtx, v.GetString())
```

Comment:
Can it be an expression like `date_sub(now(), interval 2 seconds)`? What does it look like in spanner?

Reply:
according to https://github.com/pingcap/parser/pull/610/files#r341901887 , the timestamp `Expression` for exact timestamp can be evaluated or converted to `DATETIME` or `TIMESTAMP`